relay-server/src/actors/upstream.rs

```diff
@@ -133,16 +190,20 @@ impl UpstreamRelay {
             .map_err(UpstreamRequestError::SendFailed)
             .and_then(|response| match response.status() {
                 StatusCode::TOO_MANY_REQUESTS => {
-                    let retry_after = response
-                        .headers()
+                    let headers = response.headers();
+                    let retry_after = headers
                         .get(header::RETRY_AFTER)
-                        .and_then(|v| v.to_str().ok())
-                        .and_then(|s| s.parse::<f64>().ok())
-                        .unwrap_or(0f64)
-                        .max(0f64)
-                        .ceil();
+                        .and_then(|v| v.to_str().ok());
+
+                    let rate_limits = headers
```

Comment:
Make sure you can deal with multiple headers here, as proxies are allowed to split `Foo: a,b` into `Foo: a\nFoo: b`

Reply:
Excellent point, this will require to collect the strings. Thanks!

Reply:
Alright, the fix in https://github.com/getsentry/relay/pull/503/commits/331df576f28fc28d789806ef0c34e608194fbb13 should do.